nixos/modules/services/networking/netbird/coturn.nix

```diff
@@ -0,0 +1,166 @@
+    };
+
+    user = mkOption {
+      type = str;
+      default = "netbird";
+      description = ''
+        The username used by netbird to connect to the coturn server.
+      '';
+    };
+
+    password = mkOption {
+      type = nullOr str;
+      default = null;
+      description = ''
+        The password of the user used by netbird to connect to the coturn server.
+      '';
+    };
+
+    passwordFile = mkOption {
+      type = nullOr path;
+      default = null;
+      description = ''
+        The path to a file containing the password of the user used by netbird to connect to the coturn server.
+      '';
+    };
+
+    port = mkOption {
+      type = port;
+      default = 3478;
+      description = ''
+        The port used by coturn to listen to traffic.
+      '';
+    };
+
+    minPort = mkOption {
+      type = port;
+      default = 49152;
+      description = "Lower bound of UDP relay endpoints.";
+    };
+
+    maxPort = mkOption {
+      type = port;
+      default = 65535;
+      description = "Lower bound of UDP relay endpoints.";
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      assertions = [
+        {
+          assertion = (cfg.password == null) != (cfg.passwordFile == null);
+          message = "Exactly one of `password` or `passwordFile` must be given for the coturn setup.";
+        }
+      ];
+
+      services.coturn =
+        {
+          enable = true;
+
+          realm = cfg.domain;
+          lt-cred-mech = true;
+          no-cli = true;
+
+          min-port = cfg.minPort;
+          max-port = cfg.maxPort;
+
+          extraConfig = ''
+            fingerprint
+            user=${cfg.user}:${if cfg.password != null then cfg.password else "@password@"}
+            no-software-attribute
+          '';
+        }
+        // (optionalAttrs cfg.useNginxCertificates (
+          let
+            dir = config.security.acme.certs.${cfg.domain}.directory;
+          in
+          {
+            cert = "${dir}/fullchain.pem";
+            pkey = "${dir}/key.pem";
+          }
+        ));
+
+      systemd.services.coturn = optionalAttrs (cfg.passwordFile != null) {
+        preStart = mkAfter ''
+          ${getExe pkgs.replace-secret} @password@ ${cfg.passwordFile} /run/coturn/turnserver.cfg
+        '';
+      };
+
+      networking.firewall = {
+        allowedUDPPorts = [
+          cfg.port
+          (cfg.port + 1)
+          5349
+          5350
+        ];
+
+        allowedTCPPorts = [
+          cfg.port
+          (cfg.port + 1)
+        ];
+
+        allowedUDPPortRanges = [
+          {
+            from = cfg.minPort;
+            to = cfg.maxPort;
+          }
+        ];
+      };
+    }
+
+    (mkIf cfg.useNginxCertificates {
+      users.users.nginx.extraGroups = [ "turnserver" ];
+
+      # share certs with coturn and restart on renewal
+      security.acme.certs.${cfg.domain} = {
+        group = "turnserver";
```

Comment:
I did it with `LoadCredential` instead of changing the group, the change to the coturn module is the same as for the `passwordFile`, so it was straightforward